pkg/reader/ccip.go

```diff
@@ -1439,5 +1439,26 @@ func (r *ccipChainReader) GetMedianDataAvailabilityGasConfig(
 	return daConfig, nil
 }
 
+func (r *ccipChainReader) GetLatestPriceSeqNr(ctx context.Context) (uint64, error) {
+	if err := validateExtendedReaderExistence(r.contractReaders, r.destChain); err != nil {
+		return 0, fmt.Errorf("validate dest=%d extended reader existence: %w", r.destChain, err)
+	}
+	var latestSeqNr uint64
+	err := r.contractReaders[r.destChain].ExtendedGetLatestValue(
+		ctx,
+		consts.ContractNameOffRamp,
+		consts.MethodNameGetLatestPriceSequenceNumber,
+		primitives.Finalized,
```

Comment:
Isn't that wrong? If you get only for finalized state here, you don't get information about increasing sequence numbers, therefore you will keep spamming until finalization is reached (around 15 minutes on Ethereum)